packages/oasgraph/src/schema_builder.ts

```diff
@@ -472,100 +472,111 @@ function createFields({
+          if (typeof argsFromLink === 'object') {
+            dynamicParams = dynamicParams.filter(p => {
+              // here, we know argsFromLink is present:
+              argsFromLink = argsFromLink as Object
+              return typeof argsFromLink[p.name] === 'undefined'
+            })
+          }
 
-        // get args for link
-        const args = getArgs({
-          parameters: dynamicParams,
-          operation,
-          data,
-          oass
-        })
+          // get resolve function for link
+          const linkResolver = getResolver({
+            operation: linkedOp,
+            argsFromLink: Oas3Tools.beautifyObjectKeys(argsFromLink) as {
+              [key: string]: string
+            },
+            data,
+            baseUrl: data.options.baseUrl
+          })
 
-        /**
-         * get response object type
-         * use the reference here
-         * OT will be built up some other time
-         */
-        const resObjectType = linkedOp.responseDefinition.ot
+          // get args for link
+          const args = getArgs({
+            parameters: dynamicParams,
+            operation,
+            data,
+            oass
+          })
 
-        let description = link.description
+          /**
+           * get response object type
+           * use the reference here
+           * OT will be built up some other time
+           */
+          const resObjectType = linkedOp.responseDefinition.ot
 
-        if (typeof description !== 'string') {
-          description = 'No description available.'
-        }
+          let description = link.description
 
-        if (oass.length === 1) {
-          description += `\n\nEquivalent to ${linkedOp.method.toUpperCase()} ${
-            linkedOp.path
-          }`
-        } else {
-          description += `\n\nEquivalent to ${
-            operation.oas.info.title
-          } ${linkedOp.method.toUpperCase()} ${linkedOp.path}`
-        }
+          if (typeof description !== 'string') {
+            description = 'No description available.'
+          }
+
+          if (oass.length === 1) {
+            description += `\n\nEquivalent to ${linkedOp.method.toUpperCase()} ${
+              linkedOp.path
+            }`
+          } else {
+            description += `\n\nEquivalent to ${
+              operation.oas.info.title
+            } ${linkedOp.method.toUpperCase()} ${linkedOp.path}`
+          }
 
-        // finally, add the object type to the fields (using sanitized field name)
-        const saneLinkKey = Oas3Tools.beautifyAndStore(linkKey, data.saneMap)
-        // TODO: check if fields already has this field name
-        fields[saneLinkKey] = {
-          type: resObjectType,
-          resolve: linkResolver,
-          args,
-          description
+          // finally, add the object type to the fields (using sanitized field name)
+          Oas3Tools.beautifyAndStore(saneLinkKey, data.saneMap)
```

Comment:
We don't need to beautify anymore because the links should already have sanitized keys. I kept it here for now because I'm not sure how the store will affect other parts of the code.